Finished translation of Fragment page (#460)
reactjs/uk.react.dev: src/content/reference/react/Fragment.md

<Intro>

> `<Fragment>`, часто використовується за допомогою синтаксису `<>...</>`, дозволяє групувати елементи без обгорткового вузла.

Reviewer: Ніколи не чув на практиці "обгортковий вузол". Можливо краще замінити на "елементу-обгортки"? Чи чогось по типу цього?

---

## Опис {/*reference*/}

### `<Fragment>` {/*fragment*/}

> Огорніть елементи у `<Fragment>` щоб групувати їх разом в ситуаціях, коли вам потрібен один елемент. Групування елементів у `Fragment` не впливає на результуючий DOM; це так само, як і якщо елементи не були згруповані. Переважно, порожній тег JSX `<></>` є скороченням для `<Fragment></Fragment>`.

Reviewer 2: ```suggestion
Огорніть елементи у `<Fragment>` щоб групувати їх разом в ситуаціях, коли вам потрібен один елемент. Групування елементів у `Fragment` не впливає на результуючий DOM; це так само, ніби елементи і не були згруповані. Переважно, порожній тег JSX `<></>` є скороченням для `<Fragment></Fragment>`.
```
На мій погляд таке формулювання є більш читабельним

#### Пропси {/*props*/}

> - **необов’язковий** `ключ (key)`: Фрагменти, оголошені з явним синтаксисом `<Fragment>` можуть мати [ключі.](/learn/rendering-lists#keeping-list-items-in-order-with-key)

Reviewer 2: ```suggestion
- **опційний** `key`: Фрагменти, оголошені з явним синтаксисом `<Fragment>` можуть мати [ключі.](/learn/rendering-lists#keeping-list-items-in-order-with-key)
```
_key_ повинен залишитися незмінним в цьому контексті. Ми ж не станемо перекладати HTML атрибути, наприклад.

}
```

> Фрагменти є корисним інструментом, оскільки групування елементів з Фрагментом не має впливу на розташування або стилізацію, на відміну від огортання елементів в інший контейнер, такий як DOM-елемент. Якщо ви перевірите цей приклад за допомогою інструментів браузера, ви побачите, що всі DOM-вузли `<h1>` і `<p>` відображаються як сусідні елементи без обгорток навколо них:

Reviewer 2: ```suggestion
Фрагменти є корисним інструментом, оскільки групування елементів з `Fragment` не має впливу на розташування або стилізацію, на відміну від огортання елементів в інший контейнер, такий як DOM-елемент. Якщо ви перевірите цей приклад за допомогою інструментів браузера, ви побачите, що всі DOM-вузли `<h1>` і `<p>` відображаються як сусідні елементи без обгорток навколо них:
```

}
```

> Зазвичай вам не потрібно цього робити, якщо вам не треба [передавати ключ у ваш `Fragment`.](#rendering-a-list-of-fragments)

Reviewer 2: ```suggestion
Зазвичай вам не потрібно цього робити, якщо вам не треба [передавати `key` у ваш `Fragment`.](#rendering-a-list-of-fragments)
```

Див. suggestion №2

</DeepDive>

---

### Присвоєння змінній декількох елементів {/*assigning-multiple-elements-to-a-variable*/}

> Як і будь-який інший елемент, фрагменти можна присвоювати змінним, передавати їх як пропси тощо:

Reviewer 2: ```suggestion
Як і будь-який інший елемент, `Fragment` елементи можна присвоювати змінним, передавати їх як пропси тощо:
```

---

### Відображення списку фрагментів {/*rendering-a-list-of-fragments*/}

> Ось ситуація, коли вам потрібно явно написати `Fragment`, замість використання синтаксису `<></>`. Якщо ви [рендерите кілька елементів в циклі](/learn/rendering-lists), вам потрібно призначити `key` кожному елементу. Якщо елементи всередині циклу є Фрагментами, то вам потрібно використовувати звичайний синтаксис елементів JSX, щоб мати змогу вказати атрибут `key`:

Reviewer 2: ```suggestion
Ось ситуація, коли вам потрібно написати `Fragment` явно, замість використання синтаксису `<></>`. Якщо ви [рендерите кілька елементів в циклі](/learn/rendering-lists), вам потрібно призначити `key` кожному елементу. Якщо елементи всередині циклу є Фрагментами, то вам потрібно використовувати звичайний синтаксис елементів JSX, щоб мати змогу вказати атрибут `key`:
```

}
```

> Ви можете перевірити DOM, щоб переконатися, що навколо "дочірніх компонентів" Fragment немає огорнутих елементів:

Reviewer 2: ```suggestion
Ви можете перевірити DOM, щоб переконатися, що дочірні компоненти Fragment нічим не обгортуються:
```
wrapper elements - елементи що обгорують, а не є обгорнутими (wrapped elements)
